src/newcharacter.cpp

```diff
@@ -1354,8 +1367,60 @@ int set_skills(WINDOW *w, player *u, int &points)
         mvwprintz(w, 3, 31, points >= cost ? COL_SKILL_USED : c_ltred,
                   ngettext("Upgrading %s costs %d point", "Upgrading %s costs %d points", cost),
                   currentSkill->name().c_str(), cost);
-        fold_and_print(w_description, 0, 0, getmaxx(w_description), COL_SKILL_USED,
-                       currentSkill->description());
+
+        std::map<std::string, std::stringstream > recipes;
+        for( auto cur_recipe : recipe_dict ) {
+            auto req_skill = cur_recipe->required_skills.find(currentSkill->ident());
+            int skill = (req_skill != cur_recipe->required_skills.end()) ? req_skill->second : 0;
+
+            if ( !cur_recipe->autolearn &&
+                 ( cur_recipe->skill_used == currentSkill->ident() || skill > 0 ) &&
+                 u->has_recipe_requirements(cur_recipe) &&
+                 cur_recipe->ident.find("uncraft") == std::string::npos
+               )  {
+
+                if ( recipes[cur_recipe->skill_used.obj().name()].str() != "" ) {
```

Comment:
`cur_recipe->skill_used.obj().name()` is quite complex and repeated trice. Why not store it in a local variable.

In fact, why not store a reference to the mapped value and use that:
```C++
auto &stream = recipes[cur_recipe->skill_used.obj().name()];
if(!stream.str().empty()) {
```